substrate/frame/revive/src/lib.rs

```diff
@@ -1134,6 +1148,9 @@ where
 		let from = tx.from.unwrap_or_default();
 		let origin = T::AddressMapper::to_account_id(&from);
 
+		let exec_context =
+			if tx.gas.is_some() { ExecContext::Transaction } else { ExecContext::DryRun };
+
```

Comment:
this is only used for dry running (maybe it need to be renamed)
so the exec_context should always be DryRun
your enum should have associated data, so that you can tell whether or not a dry run has a deposit limit